core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-container-code.component.ts

```diff
@@ -45,6 +51,20 @@ export class DotContentEditorComponent implements OnInit {
         this.updateActiveTabIndex(this.getcontainerStructures.length);
     }
 
+    ngOnChanges(changes: SimpleChanges) {
+        changes.contentTypes.currentValue.map(({ id, name }: DotCMSContentType) => {
+            this.contentTypeNamesById[id] = name;
+        });
+
+        this.init();
+        this.updateActiveTabIndex(this.getcontainerStructures.length);
+        if (changes.contentTypes.currentValue.length > 0) {
+            Object.keys(this.monacoEditors).forEach((editorId) => {
+                this.monacoEditors[editorId].updateOptions({ readOnly: false });
+            });
+        }
+    }
```

Comment:
How many times this gets called in a lifecycle?

Reply:
maximum 2 time only